src/safe_apps/models.py

```diff
@@ -17,7 +17,12 @@
 
 def safe_app_icon_path(instance: "SafeApp", filename: str) -> str:
     _, file_extension = os.path.splitext(filename)
-    return f"safe_apps/{instance.app_id}/icon{file_extension}"
+    if instance.app_id is None:  # new SafeApp, app_id is not available at this point
+        safe_apps = SafeApp.objects.all().order_by("-app_id")
+        app_id = safe_apps[0].app_id + 1 if safe_apps else 1
+        return f"safe_apps/{app_id}/icon{file_extension}"
+    else:  # existing SafeApp, app_id is available
+        return f"safe_apps/{instance.app_id}/icon{file_extension}"
```

Comment:
Since `app_id` is not available at this point, this code mimics the logic of assigning a new ID to the `SafeApp`. If for any reason this calculated ID does not match the DB one (deletion race condition for example), it will be eventually fixed when the `SafeApp` gets updated.

An alternative solution would be using a random UUID instead of the `app_id` field, but I find it useful the URLs match the app ID. I'm open to suggestions though.

Reply:
After an internal sync, we've decided to take the UUID approach to compose the logo URL. (Thanks for your feedback @iamacook)

[e16544e](https://github.com/safe-global/safe-config-service/pull/1107/commits/e16544ea2040bfbeb8d643b5b8bc9456a1cc38ab)